sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/analysis/AnalysisSuite.scala

```diff
@@ -345,4 +345,42 @@ class AnalysisSuite extends AnalysisTest {
 
     assertAnalysisSuccess(query)
   }
+
+  test("SPARK-15776: test whether Divide expression's data type can be deduced correctly by " +
+    "analyzer") {
+    val testRelation = LocalRelation(AttributeReference("a", IntegerType)())
+    assertSameSchema(
```

Comment:
How about defining a tester function, and use that for the rest of the testing? Something like:
```scala
def assertDivideReturnType(left: Expression, right: Expression, dt: DataType): Unit = {
  assertSameSchema(
   Project(Seq(Alias(sum(Divide(left, right)), "a")()), OneRowRelation),
   StructType(Seq(StructField("a", dt, true))))
}
```